lib/puppet/functions/conjur/client.rb

```diff
@@ -83,13 +79,10 @@ def http
       end
 
       def variable_value account, id, token = nil
-        path = case version
-          when 4
-            ['variables', URI.encode_www_form_component(id), 'value']
-          when 5
-            raise ArgumentError, "account is required for v5" unless account
-            ['secrets', account, 'variable', ERB::Util.url_encode(id)]
-               end.join('/')
+        raise ArgumentError, "account is required for v5" unless account
+
+        path = ['secrets', account, 'variable', ERB::Util.url_encode(id)].join('/')
```

Comment:
My Ruby knowledge is limited, but just checking... was the deletion of `.end` (before the `.join`) intentional here?

Reply:
Oh yeah this is fine. The `end` corresponded to the for the `when`s. Blocks in ruby can return data that's why it was `end.join` 🧙